### DISCUSSION TITLE
Plotting error

### DISCUSSION THREAD

**Discussion by srcgps** [insarlab/MintPy#1251] [Q&A]
Hi,

Can you please assist with the below error I encountered while running smallbaselineapp.py?

Go to work directory: /media/srcgps/Elements/SVGstack_Kiran/MINTPY
read default template file: /media/srcgps/Elements/SVGstack_Kiran/MINTPY/smallbaselineApp.cfg

******************** plot & save to pic ********************
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 velocity.h5 --dem inputs/geometryRadar.h5 --mask maskTempCoh.h5
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 temporalCoherence.h5 -c gray -v 0 1
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 maskTempCoh.h5 -c gray -v 0 1
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 inputs/geometryRadar.h5
Traceback (most recent call last):
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/smallbaselineApp.py", line 213, in <module>
    main(sys.argv[1:])
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/smallbaselineApp.py", line 208, in main
    run_smallbaselineApp(inps)
  File "/home/srcgps/tools/MintPy/src/mintpy/smallbaselineApp.py", line 1123, in run_smallbaselineApp
    app.plot_result()
  File "/home/srcgps/tools/MintPy/src/mintpy/smallbaselineApp.py", line 1061, in plot_result
    mintpy.cli.view.main(iargs)
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/view.py", line 175, in main
    obj.plot()
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1616, in plot
    plot_figure(j, self, metadata=self.atr)
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1290, in plot_figure
    data = read_data4figure(i_start, i_end, inps, metadata)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1087, in read_data4figure
    data[i - i_start, :, :] = d
    ~~~~^^^^^^^^^^^^^^^^^^^
ValueError: could not broadcast input array from shape (1395,1585) into shape (1396,1585)

thank you.

**Comment by welcome** [comment]
👋  Thanks for opening your first issue here! Please filled out the template with as much details as possible. We appreciate that you took the time to contribute!
Make sure you read our [contributing guidelines](https://github.com/insarlab/MintPy/blob/master/docs/CONTRIBUTING.md).

**Comment by yunjunz** [comment]
Please follow the [bug report template](https://github.com/insarlab/MintPy/blob/main/.github/ISSUE_TEMPLATE/bug_report.md) and fill in all the info there for better diagnosis.

**Comment by srcgps** [comment]
---
name: Bug report
about: Report a problem/bug to help us improve
title: 'Plotting error'
labels: ''
assignees: ''

---

**Description of the problem**

The smallbaselineApp.py throws a value error due to different dimensions in the images in the step - hdfeos5


**Full script that generated the error**

```
# vim: set filetype=cfg:
##------------------------------- ISCE/topsStack ----------------------##
#ssaraopt = --platform=SENTINEL-1A,SENTINEL-1B -r 128 -f 589,590,591,592,593  -e 2017-07-01
#sentinelStack.boundingBox      = '13.30 13.37 -61.22 -61.12'
#sentinelStack.subswath         = 3     # comment
#sentinelStack.numConnections   = 5      # comment
#sentinelStack.azimuthLooks     = 5       # comment
#sentinelStack.rangeLooks       = 15      # comment
#sentinelStack.filtStrength     = 0.2     # comment
#sentinelStack.unwMethod        = snaphu  # comment
#sentinelStack.coregistration   = auto    # comment
#subset.y0:y1,x0:x1 = 1150:1600,1070:1670


##-------------------------------- MintPy -----------------------------##
########## 1. Load Data (--load to exit after this step)
## load_data.py -H to check more details and example inputs.
mintpy.load.processor        = isce
##---------for ISCE only:
mintpy.load.metaFile         = /media/srcgps/Elements/SVGstack_Kiran/reference/IW*.xml
mintpy.load.baselineDir      = /media/srcgps/Elements/SVGstack_Kiran/baselines
##---------interferogram datasets:
mintpy.load.unwFile          = /media/srcgps/Elements/SVGstack_Kiran/merged/interferograms/*/filt_*.unw
mintpy.load.corFile          = /media/srcgps/Elements/SVGstack_Kiran/merged/interferograms/*/filt_*.cor
mintpy.load.connCompFile     = /media/srcgps/Elements/SVGstack_Kiran/merged/interferograms/*/filt_*.unw.conncomp
##---------geometry datasets:
mintpy.load.demFile          = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/hgt.rdr
mintpy.load.lookupYFile      = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/lat.rdr
mintpy.load.lookupXFile      = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/lon.rdr
mintpy.load.incAngleFile     = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/los.rdr
mintpy.load.azAngleFile      = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/los.rdr
mintpy.load.shadowMaskFile   = /media/srcgps/Elements/SVGstack_Kiran/merged/geom_reference/shadowMask.rdr
mintpy.load.waterMaskFile    = /media/srcgps/Elements/SVGstack_Kiran/watermask/waterMask.rdr
mintpy.troposphericDelay.method = height_correlation
#mintpy.subset.yx = 200:1000,250:750
#mintpy.reference.maskFile    = no
#mintpy.reference.yx         = 634,185
mintpy.reference.lalo         = auto
#mintpy.unwrapError.method = no

#mintpy.topographicResidual.stepDate = 2021  #eruption dates
#mintpy.deramp                       = linear

```


**Full error message**

```
******************** step - hdfeos5 ********************
save time-series to HDF-EOS5 format is OFF.

******************** plot & save to pic ********************
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 velocity.h5 --dem inputs/geometryRadar.h5 --mask maskTempCoh.h5
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 temporalCoherence.h5 -c gray -v 0 1
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 maskTempCoh.h5 -c gray -v 0 1
view.py --dpi 150 --noverbose --nodisplay --update --memory 4.0 inputs/geometryRadar.h5
Traceback (most recent call last):
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/smallbaselineApp.py", line 213, in <module>
    main(sys.argv[1:])
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/smallbaselineApp.py", line 208, in main
    run_smallbaselineApp(inps)
  File "/home/srcgps/tools/MintPy/src/mintpy/smallbaselineApp.py", line 1123, in run_smallbaselineApp
    app.plot_result()
  File "/home/srcgps/tools/MintPy/src/mintpy/smallbaselineApp.py", line 1061, in plot_result
    mintpy.cli.view.main(iargs)
  File "/home/srcgps/tools/MintPy/src/mintpy/cli/view.py", line 175, in main
    obj.plot()
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1616, in plot
    plot_figure(j, self, metadata=self.atr)
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1290, in plot_figure
    data = read_data4figure(i_start, i_end, inps, metadata)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/srcgps/tools/MintPy/src/mintpy/view.py", line 1087, in read_data4figure
    data[i - i_start, :, :] = d
    ~~~~^^^^^^^^^^^^^^^^^^^
ValueError: could not broadcast input array from shape (1395,1585) into shape (1396,1585)
```


**System information**

* Operating system: Linux
* Python environment: conda
* MintPy version: MintPy version 1.5.1.post23, date 2023-04-19
* InSAR processor/product: isce2
* Your custom / default template file (if the bug is related to a specific dataset):

**Comment by yunjunz** [comment]
1. Your mintpy version is very old, please update to the latest version
2. The reported inconsistent size in the geometryRadar.h5 file is likely due to the water mask file you specified, please follow [this notebook](https://github.com/insarlab/MintPy-tutorial/blob/main/applications/water_mask.ipynb) to prepare one.
